src/scripting/KM_ScriptingStates.pas

```diff
@@ -507,6 +511,31 @@ function TKMScriptStates.GameTime: Cardinal;
 end;
 
 
+//* Version: 7000+
+//* Returns an array with IDs of teams for all players
+//* Result: Array of team IDs
+function TKMScriptStates.GetAllTeams: TIntegerArray;
+var
+  I, NetIndex: Integer;
+begin
+  try
+    SetLength(Result, 0);
+    for I := 1 to gGame.Networking.NetPlayers.Count do
+    begin
+      NetIndex := gGame.Networking.NetPlayers.PlayerIndexToLocal(I);
+      SetLength(Result, Length(Result) + 1);
```

Comment:
No need to resize the array every loop (inefficient), just set the size once at the start.